subprojects/core/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheService.java

```diff
@@ -152,17 +167,11 @@ public void run() {
     }
 
     @Override
-    public void load(final BuildCacheKey key, final Action<? super File> reader) {
-        persistentCache.withFileLock(new Factory<Void>() {
+    public void allocateTempFile(final BuildCacheKey key, final Action<? super File> action) {
+        persistentCache.useCache(new Runnable() {
```

Comment:
I had that originally, then took it out because I thought there would be concerns about relic tmp files.

Will fix it. It's easy.